### PR TITLE
RpcConnector finished

### PR DESCRIPTION
**Pull request by pavolpluta**

### CODE REVIEW

File: src/perun/micro_services/adapters/RpcConnector.py [hunk 1]
```diff
@@ -19,12 +31,50 @@ def __init__(self, rpc_url='', user='', password=''):
...
+        body = buffer.getvalue()
+        result_json = body.decode('utf-8')
+        result = json.loads(result_json)
+
+        if 'errorId' in result.keys():
+            raise Exception(f"Exception from Perun: {result['message']}")
+
+        response_time = round(end_time - start_time, 3)
+        logger.debug(
+            f'GET call {uri} with params: {params_query}, response: {json} in: {response_time} s.'
```
Comment (vyskocilpavel) [inline src/perun/micro_services/adapters/RpcConnector.py:67]: json is not object with Perun response. You should use result.

Comment (vyskocilpavel) [reply src/perun/micro_services/adapters/RpcConnector.py:67]: And the response time is not in seconds but in milliseconds.

File: src/perun/micro_services/adapters/RpcConnector.py [hunk 2]
```diff
@@ -50,16 +100,15 @@ def post(self, manager, method, params=None):
...
-        response_time = end_time - start_time
-        logger.error(
-            "POST call {} with params {}, response {} in {} ms".format(
-                uri,
-                params_json,
-                result,
-                round(response_time * 1000)
-            )
-        )
+            raise Exception(f"Exception from Perun: {result['message']}")
```
Comment (vyskocilpavel) [inline src/perun/micro_services/adapters/RpcConnector.py:103]: Use f' instead of f"

File: src/perun/micro_services/adapters/RpcConnector.py [hunk 3]
```diff
@@ -50,16 +100,15 @@ def post(self, manager, method, params=None):
...
+
+        response_time = round(end_time - start_time, 3)
+        logger.debug(
+            f'POST call {uri} with params: {params_json}, response: {result} in: {response_time} s')
```
Comment (vyskocilpavel) [inline src/perun/micro_services/adapters/RpcConnector.py:107]: The same thing as in get method. The response time is in milliseconds.

File: src/perun/micro_services/adapters/RpcConnector.py [hunk 4]
```diff
@@ -19,12 +31,50 @@ def __init__(self, rpc_url='', user='', password=''):
...
+        c.perform()
+        end_time = self.__microtime()
+        c.close()
+
+        body = buffer.getvalue()
+        result_json = body.decode('utf-8')
+        result = json.loads(result_json)
+
+        if 'errorId' in result.keys():
+            raise Exception(f"Exception from Perun: {result['message']}")
```
Comment (vyskocilpavel) [inline src/perun/micro_services/adapters/RpcConnector.py:63]: Use f' instead of f"

File: src/perun/micro_services/adapters/RpcConnector.py [hunk 5]
```diff
@@ -50,16 +100,15 @@ def post(self, manager, method, params=None):
...
-        )
+            raise Exception(f"Exception from Perun: {result['message']}")
+
+        response_time = round(end_time - start_time, 3)
+        logger.debug(
+            f'POST call {uri} with params: {params_json}, response: {result} in: {response_time} s')
 
         return result
+
+    # '__' at the beginning states that it is a private method
```
Comment (vyskocilpavel) [inline src/perun/micro_services/adapters/RpcConnector.py:111]: The comment can be removed.

File: src/perun/micro_services/adapters/RpcConnector.py [hunk 6]
```diff
@@ -1,15 +1,27 @@
+"""
+Provides interface to call Perun RPC.
+Note that Perun RPC should be considered as unreliable
+and authentication process should continue without connection to Perun. e.g. use LDAP instead.
+
+Example:
```
Comment (vyskocilpavel) [inline src/perun/micro_services/adapters/RpcConnector.py:6]: Remove the example.

File: src/perun/micro_services/adapters/RpcConnector.py [hunk 7]
```diff
@@ -50,16 +100,15 @@ def post(self, manager, method, params=None):
...
+
+        response_time = round(end_time - start_time, 3)
+        logger.debug(
+            f'POST call {uri} with params: {params_json}, response: {result} in: {response_time} s')
 
         return result
+
+    # '__' at the beginning states that it is a private method
+    @staticmethod
+    def __microtime():
```
Comment (vyskocilpavel) [inline src/perun/micro_services/adapters/RpcConnector.py:113]: And this method returns time in milliseconds, not in microseconds.
The better solution which is more precise is to use this function: https://docs.python.org/3/library/time.html#time.time_ns 
which return time in nanoseconds.

But this method is available from python v3.7 so we must remove python 3.6 from the supported version if we use this method.

File: src/perun/micro_services/adapters/RpcConnector.py [hunk 8]
```diff
@@ -100,15 +98,14 @@ def post(self, manager, method, params=None):
...
+            f'POST call {uri} with params: {params_json}, response: {result} in: {response_time} ms')
 
         return result
 
-    # '__' at the beginning states that it is a private method
     @staticmethod
-    def __microtime():
-        return int(round(time.time() * 1000))
+    def __millitime():
+        return time.time_ns() // 1000000
```
Comment (pavolpluta) [inline src/perun/micro_services/adapters/RpcConnector.py:111]: Don't forget to change supported Python version to >=3.7

**Pull request merged**